### ISSUE TITLE
MissingSwitchDefault for switch exception (Java 14+)

### ISSUE THREAD

**Issue by sukolenvo** [checkstyle/checkstyle#10463]
I have read check documentation: https://checkstyle.org/config_coding.html#MissingSwitchDefault
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Main.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
  <module name="TreeWalker">
    <module name="MissingSwitchDefault"/>
  </module>
</module>


/var/tmp $ cat Main.java
import java.nio.file.AccessMode;

public class Main {

  private void createPlatform() {
    String mode;
    mode = switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.44-all.jar -c config.xml Main.java
Starting audit...
[ERROR] /home/gtx/Downloads/Main.java:7:12: switch without "default" clause. [MissingSwitchDefault]
Audit done.
Checkstyle ends with 1 errors.
```
**Expected**: no errors. Following code is very similar and does not produce any errors:
```java
  private void createPlatform() {
    String mode = switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }

  private String createPlatform() {
    return switch (AccessMode.READ) {
      case READ -> "1";
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
```
Given javac won't compile non-exhaustive switch, maybe `MissingSwitchDefault` check should be no-op for switch exceptions? Sample javac error:
```bash
/var/tmp $ cat Main.java 
import java.nio.file.AccessMode;

public class Main {

  private String createPlatform() {
    return switch (AccessMode.READ) {
      case WRITE -> "2";
      case EXECUTE -> "3";
    };
  }
}

/var/tmp $ javac Main.java
Main.java:6: error: the switch expression does not cover all possible input values
    return switch (AccessMode.READ) {
           ^
1 error
```

**Comment by nmancus1** [comment]
Checkstyle does check if we are in an exhaustive switch expression (at least this is the intention), and will not log the error if so:

https://github.com/checkstyle/checkstyle/blob/2e453a571f868a83ef9bf9d3112f4168e68ac9d5/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java#L152-L160

In the two examples that do not report an error, the `LITERAL_SWITCH` token's parent is `EXPR`, which is caught by the above logic. However, the example in your bug report, `ASSIGN` is the parent of `LITERAL_SWITCH`:

```bash

[9:35:26] ➜  src java -jar checkstyle-8.44-all.jar -t Test.java
...
    |       |--VARIABLE_DEF -> VARIABLE_DEF [4:4]
    |       |   |--MODIFIERS -> MODIFIERS [4:4]
    |       |   |--TYPE -> TYPE [4:4]
    |       |   |   `--IDENT -> String [4:4]
    |       |   `--IDENT -> mode [4:11]
    |       |--SEMI -> ; [4:15]
    |       |--EXPR -> EXPR [5:9]
    |       |   `--ASSIGN -> = [5:9]
    |       |       |--IDENT -> mode [5:4]
    |       |       `--LITERAL_SWITCH -> switch [5:11]
    |       |           |--LPAREN -> ( [5:18]
    |       |           |--EXPR -> EXPR [5:29]
    |       |           |   `--DOT -> . [5:29]
    |       |           |       |--IDENT -> AccessMode [5:19]
    |       |           |       `--IDENT -> READ [5:30]
...

```

We need to revisit the logic of the `isSwitchExpression()` method to accommodate assignment expressions, not just return statements and variable declarations.

**Comment by sukolenvo** [comment]
Duplicate of https://github.com/checkstyle/checkstyle/issues/10866
